src/NuGet.Core/NuGet.Packaging/Core/comparers/PackageDependencyGroupComparer.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Frameworks;
+using NuGet.Shared;
+
+namespace NuGet.Packaging.Core
+{
+    public class PackageDependencyGroupComparer : IEqualityComparer<PackageDependencyGroup>
+    {
+        public PackageDependencyGroupComparer()
+        {
+        }
+
+        /// <summary>
+        /// Default comparer
+        /// </summary>
+        public static PackageDependencyGroupComparer Default { get; } = new PackageDependencyGroupComparer();
+
+        public bool Equals(PackageDependencyGroup x, PackageDependencyGroup y)
+        {
+            if (ReferenceEquals(x, y))
+            {
+                return true;
+            }
+
+            if (ReferenceEquals(x, null)
+                || ReferenceEquals(y, null))
+            {
+                return false;
+            }
+
+            return EqualityComparer<NuGetFramework>.Default.Equals(x.TargetFramework, y.TargetFramework)
+                && EqualityComparer<IEnumerable<PackageDependency>>.Default.Equals(x.Packages, y.Packages);
```

Comment:
I think we should probably sort the package dependencies by PackageIdentity, https://github.com/NuGet/NuGet.Client/blob/44b3089bf8b88a56df3bddfea3ebb318b7ddb8ce/build/Shared/EqualityUtility.cs#L23.